numba/tests/test_random.py

```diff
@@ -1291,6 +1294,98 @@ def test_multinomial_3_tuple(self):
             self._check_sample(n, pvals, sample)
 
 
+class TestRandomDirichlet(BaseTest):
+    alpha = np.array([1, 1, 1, 2], dtype=np.float64)
+
+    def _check_sample(self, alpha, size, sample):
+        '''Check output properties'''
+        if not size is None:
+
+            if type(size) is tuple:
+                self.assertIsInstance(sample, np.ndarray)
+                self.assertEqual(sample.shape, size + (len(alpha),))
+                self.assertEqual(sample.dtype, np.dtype(np.float64))
+                '''Check statistical properties'''
+                self.assertAlmostEqual(sample.sum(), np.prod(size), places=5)
+                for val in np.nditer(sample):
+                    self.assertGreaterEqual(val, 0)
+                    self.assertLessEqual(val, 1)
+
+            if type(size) is int:
+                self.assertIsInstance(sample, np.ndarray)
+                self.assertEqual(sample.shape, (size, len(alpha)))
+                self.assertEqual(sample.dtype, np.dtype(np.float64))
+                '''Check statistical properties'''
+                self.assertAlmostEqual(sample.sum(), size, places=5)
+                for val in np.nditer(sample):
+                    self.assertGreaterEqual(val, 0)
+                    self.assertLessEqual(val, 1)
+        else:
+            self.assertIsInstance(sample, np.ndarray)
+            self.assertEqual(sample.size, len(alpha))
+            self.assertEqual(sample.dtype, np.dtype(np.float64))
+            '''Check statistical properties'''
+            self.assertAlmostEqual(sample.sum(), 1, places=5)
+            for val in np.nditer(sample):
+                self.assertGreaterEqual(val, 0)
+                self.assertLessEqual(val, 1)
```

Comment:
The 3 blocks in here look like they could be parameterised into an expected "shape" and "statistical property" ?